lib/utils/cobs.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2024 Kelly Helmut Lord
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <zephyr/data/cobs.h>
+
+int cobs_encode(struct net_buf *src, struct net_buf *dst, uint32_t flags)
+{
+	int delimiter = COBS_FLAG_CUSTOM_DELIMITER(flags);
+
+	/* Calculate required space for worst case */
+	size_t max_encoded_size = src->len + (src->len / 254) + 2;
```

Comment:
Use your own function?

```suggestion
	size_t max_encoded_size = cobs_max_encoded_len(src->len, flags);
```